loop/files/aws.js

```diff
@@ -0,0 +1,100 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+var aws = require('aws-sdk');
+var encode = require('../utils').encode;
+var decode = require('../utils').decode;
+var isUndefined = require('../utils').isUndefined;
+
+var DEFAULT_PUBLIC_BUCKET = "room_encrypted_files";
+var CONTENT_TYPE = "application/json";
+
+function AwsDriver(options, settings) {
+  this._settings = settings || {};
+  this._publicBucket = options.publicBucket || DEFAULT_PUBLIC_BUCKET;
+  if (!/^[a-zA-Z0-9_\-]+$/.test(this._publicBucket)) {
+    throw new Error('Illegal Bucket Name: ' + this._publicBucket);
+  }
+  this._s3 = new aws.S3();
+}
+
+AwsDriver.prototype = {
+
+  /**
+   * Create or override a file.
+   *
+   * @param {String}    filename, the filename of the object to store.
+   * @param {String}    body, the content of the file to store
+   * @param {Function}  A callback that will be called once data had been
+   *                    stored.
+   **/
+  write: function(filename, body, callback) {
+    if (isUndefined(filename, "filename", callback)) return;
+    if (body === undefined) return callback(null, null);
```

Comment:
During the transition we can have room without context. In that case we want to let this works.
When the content is null, we shouldn't write on S3.

This is consistent with the read method which doesn't fail on unexisting keys, but return null instead.
Writting underfined and writting nothing is the exact same thing so we should not